pkg/download/bucket/bucket.go

```diff
@@ -0,0 +1,171 @@
+	result := make(v2.ConfigsPerType)
+	response, err := d.client.List(context.TODO())
+	if err != nil {
+		log.WithFields(field.Type("bucket"), field.Error(err)).Error("Failed to fetch all bucket definitions: %v", err)
+		return result, nil
+	}
+
+	if !response.IsSuccess() {
+		log.WithFields(field.Type("bucket"), field.Error(err)).Error("Failed to fetch all bucket definitions: %s", string(response.Data))
+		return result, nil
+	}
+
+	configs := d.convertAllObjects(projectName, response.Objects)
+	result["bucket"] = configs
+	return result, nil
+}
+
+func (d *Downloader) convertAllObjects(projectName string, objects [][]byte) []config.Config {
+	result := make([]config.Config, 0, len(objects))
+	for _, o := range objects {
+		var bucketName struct {
+			BucketName  string `json:"bucketName"`
+			DisplayName string `json:"displayName"`
+		}
+		if err := json.Unmarshal(o, &bucketName); err != nil {
+			return result
+		}
+
+		// exclude builtin bucket names
+		if strings.HasPrefix(bucketName.BucketName, "dt_") || strings.HasPrefix(bucketName.BucketName, "default_") {
+			continue
+		}
+
+		// construct config object with generated config ID
+		configID := idutils.GenerateUUIDFromString(bucketName.BucketName)
+		configCoord := coordinate.Coordinate{
+			Project:  projectName,
+			Type:     "bucket",
+			ConfigId: configID,
+		}
+
+		params := map[string]parameter.Parameter{}
+		{
+			var s string
+			var err error
+
+			s, err = getValueForAttribute(o, "bucketName")
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+			params[config.NameParameter] = &value.ValueParameter{Value: s}
+			o, err = replaceAttributeWith(o, "bucketName", config.NameParameter)
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+
+			s, err = getValueForAttribute(o, "displayName")
+			if err != nil {
+				log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+				continue
+			}
+			if s != "" {
+				params["displayName"] = &value.ValueParameter{Value: s}
+				o, err = replaceAttributeWith(o, "displayName", "displayName")
+				if err != nil {
+					log.WithFields(field.Coordinate(coordinate.Coordinate{Project: projectName, Type: "bucket", ConfigId: configID}), field.Error(err)).Warn("Failed to get configuration for %v (%s): %v", configID, "bucket", err)
+					continue
+				}
+			}
+
+		}
+
+		originObjectID := fmt.Sprintf("%s_%s", configCoord.Project, configCoord.ConfigId)
```

Comment:
bucketName is the unique ID for the bucket API, so that should be the originObjectID:
```suggestion
		originObjectID := bucketName.bucketName
```